### DISCUSSION TITLE
How to use Cubit to create a 2D or QUAD9 Element

### DISCUSSION THREAD

**Discussion by alexchen4ai** [idaholab/moose#18742] [Q&A Meshing] [answered]
Hi, I am using cubit to create the mesh for Moose. I download the Cubit-learn version and it seems that there is no option for 2D mesh generation. Also, I can't find the option to adjust the type of elements. For example, use QUAD9 element so that we could use second order interpolation. Can anyone help with this?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Here are examples using the python API to generate a 2D mesh (there might be better ones in the documentation btw)
https://github.com/idaholab/virtual_test_bed/tree/main/msfr/mesh
https://github.com/idaholab/virtual_test_bed/blob/main/pbfhr/meshes/generate_core_with_reflectors.py

And selecting quad9 is done here
https://cubit.sandia.gov/15.5/help_manual/WebHelp/finite_element_model/exodus/exodus2_file_specification.htm

Guillaume

**Reply by alexchen4ai** [reply]
Thank you. This works perfectly.
